cookbook/agent_concepts/tool_concepts/custom_tools/confirmation_required.py

```diff
@@ -0,0 +1,71 @@
+"""🤝 Human-in-the-Loop: Adding User Confirmation to Tool Calls
+
+This example shows how to implement human-in-the-loop functionality in your Agno tools.
+It shows how to:
+- Add pre-hooks to tools for user confirmation
+- Handle user input during tool execution
+- Gracefully cancel operations based on user choice
+
+Some practical applications:
+- Confirming sensitive operations before execution
+- Reviewing API calls before they're made
+- Validating data transformations
+- Approving automated actions in critical systems
+
+Run `pip install openai httpx rich agno` to install dependencies.
+"""
+
+import json
+
+from agno.run.response import RunEvent, RunResponse
+import httpx
+from agno.agent import Agent
+from agno.models.openai import OpenAIChat
+from agno.tools import tool
+
+
+@tool(requires_confirmation=True)
+def get_top_hackernews_stories(num_stories: int) -> str:
+    """Fetch top stories from Hacker News.
+
+    Args:
+        num_stories (int): Number of stories to retrieve
+
+    Returns:
+        str: JSON string containing story details
+    """
+    # Fetch top story IDs
+    response = httpx.get("https://hacker-news.firebaseio.com/v0/topstories.json")
+    story_ids = response.json()
+
+    # Yield story details
+    all_stories = []
+    for story_id in story_ids[:num_stories]:
+        story_response = httpx.get(
+            f"https://hacker-news.firebaseio.com/v0/item/{story_id}.json"
+        )
+        story = story_response.json()
+        if "text" in story:
+            story.pop("text", None)
+        all_stories.append(story)
+    return json.dumps(all_stories)
+
+
+agent = Agent(
+    model=OpenAIChat(id="gpt-4o-mini"),
+    tools=[get_top_hackernews_stories],
+    markdown=True,
+    debug_mode=True
+)
+
+agent.run("Fetch the top 2 hackernews stories")
+if agent.run_response.event == RunEvent.tool_calls_paused.value:
```

Comment:
we could wrap this check in `Agent` or `RunResponse` so it's more declarative e.g.:
```python
if agent.is_paused:
```